lib/engine/game/g_18_pa/game.rb

```diff
@@ -145,23 +143,89 @@ class Game < Game::Base
                   },
         ].freeze
 
+        SCRANTON_HEX = 'G12'
+        SCRANTON_MARKER_ICON = 'mine'
+        SCRANTON_MARKER_COST = 40
+        DOUBLING_TOKEN_CORPS = %w[B&A ERIE PRR].freeze
+        MINOR_UPGRADES = %w[yellow green].freeze
+
+        def new_auction_round
+          Engine::Round::Auction.new(self, [
+            Engine::Step::SelectionAuction,
+          ])
+        end
+
+        def stock_round
+          Engine::Round::Stock.new(self, [
+            Engine::Step::DiscardTrain,
+            Engine::Step::Exchange,
+            Engine::Step::SpecialTrack,
+            Engine::Step::BuySellParShares,
+          ])
+        end
+
         def operating_round(round_num)
           Round::Operating.new(self, [
             Engine::Step::Bankrupt,
-            Engine::Step::Exchange,
-            Engine::Step::SpecialTrack,
             Engine::Step::SpecialToken,
-            Engine::Step::BuyCompany,
-            Engine::Step::HomeToken,
-            Engine::Step::Track,
+            G18PA::Step::Track,
             Engine::Step::Token,
             Engine::Step::Route,
-            Engine::Step::Dividend,
+            G18PA::Step::Dividend,
             Engine::Step::DiscardTrain,
             Engine::Step::BuyTrain,
-            [Engine::Step::BuyCompany, { blocks: true }],
           ], round_num: round_num)
         end
+
+        def setup
+          @scranton_marker_ability = Engine::Ability::Description.new(type: 'description', description: 'Scranton Token')
+
+          # place the home station for all corporations and minors except NYC.
+          @corporations.each do |corporation|
+            next if corporation.id == 'NYC'
+
+            tile = hex_by_id(corporation.coordinates)&.tile
```

Comment:
You don't need the safe navigation operator. `hex_by_id` isn't going to return nil here, and if it did the next line of code would fail anyway.
```suggestion
            tile = hex_by_id(corporation.coordinates).tile
```

Reply:
fixed